app/controllers/users_controller.rb

```diff
@@ -1,31 +1,74 @@
 class UsersController < ApplicationController
+  before_action :logged_in_user, :correct_user, only: %i(edit update)
+  before_action :admin_user, only: %i(destroy)
+  before_action :load_user, except: %i(index new create)
+
+  def index
+    @users = User.page(params[:page]).per Settings.quantity_per_page
+  end
 
   def new
     @user = User.new
   end
 
-  def show
-    @user = User.find_by id: params[:id]
-    unless @user
-      flash[:error] = t(".user_not_found")
-      redirect_to root_path
-    end
-  end
+  def show; end
 
   def create
     @user = User.new user_params
     if @user.save
+      log_in @user
       flash[:success] = t(".welcome_to_the_sample_app")
       redirect_to @user
     else
       render :new
     end
   end
 
-  private
+  def edit; end
+
+  def update
+     if @user.update user_params
+      flash[:success] = t(".profile_updated")
+      redirect_to @user
+    else
+      render :edit
+    end
+  end
 
-    def user_params
-      params.require(:user).permit(:name, :email, :password,
-                                   :password_confirmation)
+  def destroy
+    if @user.destroy
+    else
+      flash[:success] = t(".user_deleted")
+      redirect_to users_url
```

Comment:
users_url khác với  users_path như thế nào?

Reply:
users_url # => http://localhost:3000/users
users_path  # => /users
users_url và users_path  là 2 helper được sinh ra :
 _url tạo một chuỗi chứa toàn bộ URL, trong khi _path tạo một chuỗi chứa đường dẫn tương đối từ thư mục gốc của ứng dụng( "/")